pyEpiabm/pyEpiabm/routine/file_population_config.py

```diff
@@ -96,13 +100,21 @@ def make_pop(input_file: str, random_seed: int = None, time: float = 0):
 
             # Raise error if microcell exists, then create new one
             for microcell in cell.microcells:
-                if microcell.id == line["microcell"]:
+
+                # The microcell id is a string separated by periods, so we
+                # must find the part which matches the csv input
+                id_parts = microcell.id.split(".")
+                if id_parts[-1] == microcell_id_csv:
                     raise ValueError(f"Duplicate microcells {microcell.id}"
                                      + f" in cell {cell.id}")
```

Comment:
Are we just iterating over the the microcells to check whether the new one is a duplicate off an existing one? If so, it would be much simpler to generate a list of IDs `[microcell.id for microcell in cell.microcells]` and then logically test if the proposed ID is already in this list. Let me know if you need help with this test. It will also be faster than this explicit list.

Reply:
This can be changed, shouldn't be hard, but we'll still need to split the string

Reply:
You can do this in the list comp too:

```
microcell_ids = [microcell.id.split('.')[-1] for microcell in cell.microcells]
```

Reply:
I actually ended up changing the approach to make it easier, by setting
`microcell_id_csv = cell_id_csv + "." + str(int(line["microcell"]))`
on line 90.